src/main/kotlin/io/github/enyason/predictions/TaskExt.kt

```diff
@@ -0,0 +1,55 @@
+import io.github.enyason.domain.models.Prediction
+import io.github.enyason.domain.models.PredictionStatus.*
+import io.github.enyason.io.github.enyason.predictions.Task
+import io.github.enyason.predictions.PredictionsApi
+import kotlinx.coroutines.delay
+import kotlin.coroutines.cancellation.CancellationException
+
+/**
+ * Calling [await] on a [Task] polls <a href="https://replicate.com">Replicate</a> until the prediction status becomes
+ * succeeded. See [io.github.enyason.domain.models.PredictionStatus.SUCCEEDED]
+ * [delayInMillis] is the time delay in milliseconds before another request to the server is triggered
+ */
+suspend fun Task<Prediction>.await(delayInMillis: Long = 2000): Prediction {
+    if (isComplete) {
+        val e = exception
+        return if (e == null) {
+            if (isCanceled) {
+                throw CancellationException("Task $this was cancelled.")
+            } else {
+                result as Prediction
+            }
+        } else {
+            throw e
+        }
+    }
+
+    val predictionsApiService = (this.service as? PredictionsApi) ?: throw Exception("PredictionsApi service is null")
+
+
+    var prediction = this.result
+    val predictionId = result?.id
+    if (prediction == null) {
+        throw Exception("Prediction object is null")
+    } else if (predictionId == null) {
+        throw Exception("Prediction ID is null")
```

Comment:
We already have a validator for predictionId, is it possible to remove this block and let that validator handle null checks too?